### ISSUE TITLE
Try with nushell

### ISSUE THREAD

**Issue by blaggacao** [gytis-ivaskevicius/nix-reinventing-the-wheel#1]
I just like the idea.

Using nushell's (cumulative, almost devshell-like) module system could give us a good way to "load" build phases onto a normal shell REPL (although be it not bash).

Same could be done with ruby ofc, but it wasn't designed as a shell, originally.

**Comment by blaggacao** [comment]
These things here: https://www.nushell.sh/book/overlays.html#basics

Meaning semantics like the following:

```console
> overlay list
──┬──────
 0 │ stdenv
 1 │ python
 2 │ conda
──┴──────

> build
unpack ...
patch ...
[ ... ]
build ...
```

**Comment by gytis-ivaskevicius** [comment]
Yes, I am well aware of nushell and I did in fact consider using it BUT it would never get approved by nixpkgs due to the required bootstrap process (you require to bootstrap rust to compile nutshell)

Other honorable mentions: Oil shell, lua, janet
